pkg/resolution/resolver/http/resolver.go

```diff
@@ -144,15 +159,33 @@ func populateDefaultParams(ctx context.Context, params []pipelinev1.Param) (map[
 
 	var missingParams []string
 
-	if _, ok := paramsMap[urlParam]; !ok {
-		missingParams = append(missingParams, urlParam)
+	if _, ok := paramsMap[urlParamType]; !ok {
+		missingParams = append(missingParams, urlParamType)
 	} else {
-		u, err := url.ParseRequestURI(paramsMap[urlParam])
+		u, err := url.ParseRequestURI(paramsMap[urlParamType])
 		if err != nil {
-			return nil, fmt.Errorf("cannot parse url %s: %w", paramsMap[urlParam], err)
+			return nil, fmt.Errorf("cannot parse url %s: %w", paramsMap[urlParamType], err)
 		}
 		if u.Scheme != "http" && u.Scheme != "https" {
-			return nil, fmt.Errorf("url %s is not a valid http(s) url", paramsMap[urlParam])
+			return nil, fmt.Errorf("url %s is not a valid http(s) url", paramsMap[urlParamType])
+		}
+	}
+
+	if username, ok := paramsMap[httpBasicAuthUsername]; ok {
+		if _, ok := paramsMap[httpBasicAuthSecret]; !ok {
+			return nil, fmt.Errorf("missing required param %s when using %s", httpBasicAuthSecret, httpBasicAuthUsername)
+		}
+		if username == "" {
+			return nil, fmt.Errorf("value %s cannot be empty", httpBasicAuthUsername)
+		}
+	}
+
+	if secret, ok := paramsMap[httpBasicAuthSecret]; ok {
+		if _, ok := paramsMap[httpBasicAuthUsername]; !ok {
+			return nil, fmt.Errorf("missing required param %s when using %s", httpBasicAuthUsername, httpBasicAuthSecret)
+		}
+		if secret == "" {
+			return nil, fmt.Errorf("value %s cannot be empty", httpBasicAuthSecret)
```

Comment:
wonder if this is the same as:
```
username := paramsMap[httpBasicAuthUsername]
secret:=paramsMap[httpBasicAuthSecret]

if username!="" && secret == ""{
return nil, fmt.Errorf("value %s cannot be empty", httpBasicAuthSecret)
}

if username=="" && secret != ""{
return nil, fmt.Errorf("value %s cannot be empty", httpBasicAuthUsername)
}
```

Reply:
it would not work with golangci-lint i think, because it wan that we explicitely check that the key exist in paramMap

Reply:
oh ok, thanks!